### ISSUE TITLE
Save work as draft - Allow user to save deposit that is in progress

### ISSUE THREAD

**Issue by njaffer** [mlibrary/umrdr#364]
1. Add button 'Save as Draft' on Save Work page
2. The draft work's visibility is set to 'private'
3. Display works with 'private' visibility as 'drafts'

**Comment by grosscol** [comment]
Some additional considerations:
* Save as Draft when Work has already been saved normally.
* Visibility of FileSets when attaching to a Draft Work.
* Clearly stating that Save will remove draft status.
* Direct access of Draft Work by link
* Direct access of Draft Work's FileSets.
* Behavior of Adding Draft Work to Collection.
* Behavior of granting edit permission of Draft Work to another User

**Comment by njaffer** [comment]
As per discussion with @grosscol, here are other possibilities:

```njaffer [2:52 PM]  
I have some ideas about 'save as a draft' option that I wanted to discuss with you. Are you in office today?

grosscol [2:52 PM]  
I won't be in hatcher today.

[2:53]  
The options are embargo, visibility, or roll our own?

njaffer [2:54 PM]  
I am thinking of roll our own

[2:54]  
The thing with visibility or embargo is that we will have to make sure that it does not break anywhere else

grosscol [2:54 PM]  
Hmmm.

[2:55]  
True. But that is the case with any option.

njaffer [2:55 PM]  
and what if later they do want to do private visibility like they have started talking about collection private visibility

grosscol [2:55 PM]  
The danger with repurposing is different than with rolling our own.

[2:56]  
Then the project would have to be re thought entirely.

njaffer [2:56 PM]  
my thoughts on 'roll our own' is to add a flag field in model, that would save  isDraft state (edited)

grosscol [2:56 PM]  
But that is a good point

[2:57]  
To the models themselves? That could work. The meaning of the predicates would be "this is incomplete and not suitable for dissemination" ?

njaffer [2:58 PM]  
Yep to the generic work model.

grosscol [3:00 PM]  
That's an interesting avenue of thought.  I don't have the bandwidth to follow that completely right now. Thursday afternoon at the earliest.

[3:00]  
Add this option to the ticket discussion so we can track it.```

**Comment by amyneeser** [comment]
Use case: I am working with a researcher who wants an assistant to upload a work and save it as a draft so they can review it before it goes live.

**Comment by njaffer** [comment]
Met with Colin & Seth and this is the plan:
- Use 'Visibility=Private' for 'isDraft'. No extra work when viewing all works
- No validation changes. The Save as Draft option will still require the user to add all the required fields
- We will explore more on 'One behalf of' option 
- Highlighted work: Check if this would be affected
- Change My work to show visibility tag [Draft]
- On New Work and Edit Work page, do not show 'Save as draft' checkbox if the visibility=open

**Comment by njaffer** [comment]
Done - can be closed once sufia7_update branch is merged into master
